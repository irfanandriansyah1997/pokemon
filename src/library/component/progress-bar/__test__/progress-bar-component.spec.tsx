import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ProgressBar from '@/library/component/progress-bar';

describe(`Testing Progress Bar`, () => {
  describe(`Testing With Snapshot`, () => {
    it(`Testing With Value Under 50%`, () => {
      const ui = shallow(<ProgressBar maxValue={100} value={45} />);

      expect(toJson(ui)).toMatchSnapshot();
    });

    it(`Testing With Value Under 75%`, () => {
      const ui = shallow(<ProgressBar maxValue={100} value={70} />);

      expect(toJson(ui)).toMatchSnapshot();
    });

    it(`Testing With Value Greater Than 75%`, () => {
      const ui = shallow(<ProgressBar maxValue={100} value={90} />);

      expect(toJson(ui)).toMatchSnapshot();
    });
  });

  describe(`Testing Progress Bar`, () => {
    it(`Testing With Value Under 50%`, () => {
      const ui = mount(<ProgressBar maxValue={100} value={45} />);

      expect(ui.html()).toMatchSnapshot();
    });

    it(`Testing With Value Under 75%`, () => {
      const ui = mount(<ProgressBar maxValue={100} value={70} />);

      expect(ui.html()).toMatchSnapshot();
    });

    it(`Testing With Value Greater Than 75%`, () => {
      const ui = mount(<ProgressBar maxValue={100} value={90} />);

      expect(ui.html()).toMatchSnapshot();
    });
  });
});
