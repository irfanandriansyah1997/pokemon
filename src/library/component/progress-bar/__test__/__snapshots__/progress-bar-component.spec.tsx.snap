// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Progress Bar Testing Progress Bar Testing With Value Greater Than 75% 1`] = `"<div class=\\"css-qkdyee\\"><div value=\\"90\\" class=\\"css-m87n6w\\"></div></div>"`;

exports[`Testing Progress Bar Testing Progress Bar Testing With Value Under 50% 1`] = `"<div class=\\"css-qkdyee\\"><div value=\\"45\\" class=\\"css-jza4d0\\"></div></div>"`;

exports[`Testing Progress Bar Testing Progress Bar Testing With Value Under 75% 1`] = `"<div class=\\"css-qkdyee\\"><div value=\\"70\\" class=\\"css-1pxnehf\\"></div></div>"`;

exports[`Testing Progress Bar Testing With Snapshot Testing With Value Greater Than 75% 1`] = `
<Styled(div)>
  <Styled(div)
    maxValue={100}
    value={90}
  />
</Styled(div)>
`;

exports[`Testing Progress Bar Testing With Snapshot Testing With Value Under 50% 1`] = `
<Styled(div)>
  <Styled(div)
    maxValue={100}
    value={45}
  />
</Styled(div)>
`;

exports[`Testing Progress Bar Testing With Snapshot Testing With Value Under 75% 1`] = `
<Styled(div)>
  <Styled(div)
    maxValue={100}
    value={70}
  />
</Styled(div)>
`;
